### DISCUSSION TITLE
Support SFCs file in AE

### DISCUSSION THREAD

**Discussion by ttquoccuong** [quasarframework/quasar#14613] [Quasar App Extensions] [answered]
Hi,
Many time, I need to create my AE with only UI components which very simple. But to write in renderFunction which make them more complicate.
Ex:
In my AE have a array of childNodes, normally we add them like
`h('div', {}, list.map(el => h('div')))`
And now I want to add 1 for every 2 childNodes. so I must be create a childComponent.
I hear this discussions but I think we should have a option to choose type of file when we create a AE.

In fact, they'll be `js`, `vue`, `jsx`.

Any idea for this?

Thanks!

**Comment by ttquoccuong** [comment]
@rstoenescu Hi, any thinking for this?

**Comment by rstoenescu** [comment, marked as answer]
The problem with SFCs is that they need to get compiled with a specific Vue version and this means that it can definitely be consumed by that specific Vue version but it's NOT guaranteed to work with any other version. For this reason, the UI kit uses `.js` files instead of `.vue`.

However, what one can do, if he/she doesn't care about UMD, is to simply write `.vue` files and deliver them as-is (uncompiled, so drop the build system in the UI kit). When those components are imported by the consuming app (q/app-webpack or q/app-vite or any other webpack or vite setup) they'll get compiled with the same Vue version and everything will be just fine.

For the reason above and for consistency, the officially UI kit will not be able to supply support for SFC **and** still have a build system in place (that takes care of commonjs/esm/UMD formats).
